fab/fabfile.py

```diff
@@ -484,31 +436,47 @@ def preindex_views():
             'Skipping preindex_views for "%s" because should_migrate = False'
         ) % env.environment)
 
-    with cd(env.code_root_preindex):
-        # update the codebase of the preindex dir
-        update_code(preindex=True)
-        # no update to env - the actual deploy will do
-        # this may break if a new dependency is introduced in preindex
-        update_virtualenv(preindex=True)
-
+    with cd(env.code_root):
         sudo((
-            'echo "%(virtualenv_root_preindex)s/bin/python '
-            '%(code_root_preindex)s/manage.py preindex_everything '
+            'echo "%(virtualenv_root)s/bin/python '
+            '%(code_root)s/manage.py preindex_everything '
             '8 %(user)s" --mail | at -t `date -d "5 seconds" '
             '+%%m%%d%%H%%M.%%S`'
         ) % env)
-        version_static(preindex=True)
+        version_static()
 
 
 @roles(ROLES_ALL_SRC)
 @parallel
-def update_code(preindex=False):
-    if preindex:
-        root_to_use = env.code_root_preindex
-    else:
-        root_to_use = env.code_root
+def update_code(use_current_release=False):
+    # If not updating current release,  we are making a new release and thus have to do cloning
+    # we should only ever not make a new release when doing a hotfix deploy
+    if not use_current_release:
+        with cd(env.code_current):
+            if files.exists(env.code_current):
+                submodules = sudo("git submodule | awk '{ print $2 }'").split()
+        with cd(env.code_root):
+            if files.exists(env.code_current):
+                local_submodule_clone = []
+                for submodule in submodules:
+                    local_submodule_clone.append('-c')
+                    local_submodule_clone.append(
+                        'submodule.{submodule}.url={code_current}/.git/modules/{submodule}'.format(
```

Comment:
this is so we don't have to pull from github the entire submodule repo and we can start off where we were on the last release. it significantly speeds things, but yea does make it a bit more terse